pytest_xmlcomp/plugin.py

```diff
@@ -89,12 +89,13 @@ def collect(self):
             print("JSON Syntax Error in file %s:\n%s" % (jsonfile, error),
                   file=sys.stderr)
             return
-        for xpath, expresult in jsondata:
-            yield XPathItem(xpath, self, expresult, tree)
+        namespaces = dict(jsondata['ns'])
```

Comment:
That line can give you a KeyError. I would suggest to use the `.get()` method. To check for empty lists or dicts, you could write:

```python
namespaces = None if not jsondata.get('ns') else dict(jsondata.get('ns'))
```